From 2d279b22010ffcc8949fb0e414a9aac3738f1ece Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Wed, 4 Aug 2021 08:36:30 +0200
Subject: [PATCH] bump to version 2022-02-20

 5ca5e0b netifd: allow disabling rule/rule6 config sections
 8875960 interface-ip: add support for IPv6 prefix invalidation
 e589c05 interface-ip: use metric when looking for a route
 b54ffde main: fix hotplug script usage message
 96902e8 Revert "netifd: add devtype to ubus call"
 29e6acf netifd: add devtype to ubus call
 7ccbf08 netifd: add devtype to ubus call
 3043206 system: fix compilation with glibc 2.34
 ed71876 iprule: add support for uidrange
 fd4c9e1 system-linux: expose hw-tc-offload ethtool feature in device status dump
 3d76f2e system-linux: add wrapper function for creating link config messages
 88af2f1 system-linux: delete bridge devices using netlink
 85c3548 system-linux: create bridge devices using netlink
 136006b cmake: fix usage of implicit library and include paths
 bc0e84d netifd: interface-ip: don't set fib6 policies if ipv6 disabled
---
 package/network/config/netifd/Makefile        |  6 ++--
 .../netifd/files/lib/netifd/dhcp.script       |  1 +
 .../config/netifd/patches/003-ethtool.patch   | 31 +++++++++++++++++++
 3 files changed, 35 insertions(+), 3 deletions(-)
 create mode 100644 package/network/config/netifd/patches/003-ethtool.patch

diff --git a/package/network/config/netifd/Makefile b/package/network/config/netifd/Makefile
index cd4f8f423d..2ab574fc57 100644
--- a/package/network/config/netifd/Makefile
+++ b/package/network/config/netifd/Makefile
@@ -5,9 +5,9 @@ PKG_RELEASE:=1
 
 PKG_SOURCE_PROTO:=git
 PKG_SOURCE_URL=$(PROJECT_GIT)/project/netifd.git
-PKG_SOURCE_DATE:=2021-10-30
-PKG_SOURCE_VERSION:=8f82742ca4f47f459284f3a07323d04da72ea5f6
-PKG_MIRROR_HASH:=5e519bb1aec9bb30782213f32f19f12e874c909e42826618dd4332ded816d2fe
+PKG_SOURCE_DATE:=2022-02-20
+PKG_SOURCE_VERSION:=136006b88826feff4f0a36ffab511d1366483cf2
+PKG_MIRROR_HASH:=6358738d20e6df27b82c3bdb575fba0fdad8bef45a3c7479b93a5587c465dba4
 PKG_MAINTAINER:=Felix Fietkau <nbd@nbd.name>
 
 PKG_LICENSE:=GPL-2.0
diff --git a/package/network/config/netifd/files/lib/netifd/dhcp.script b/package/network/config/netifd/files/lib/netifd/dhcp.script
index e46005d84c..6fcf139beb 100755
--- a/package/network/config/netifd/files/lib/netifd/dhcp.script
+++ b/package/network/config/netifd/files/lib/netifd/dhcp.script
@@ -60,6 +60,7 @@ setup_interface () {
 	[ -n "$message" ]  && json_add_string message "$message"
 	[ -n "$timezone" ] && json_add_int timezone "$timezone"
 	[ -n "$lease" ]    && json_add_int leasetime "$lease"
+	[ -n "$serverid" ] && json_add_string dhcpserver "$serverid"
 	proto_close_data
 
 	proto_send_update "$INTERFACE"
diff --git a/package/network/config/netifd/patches/003-ethtool.patch b/package/network/config/netifd/patches/003-ethtool.patch
new file mode 100644
index 0000000000..6aee86283e
--- /dev/null
+++ b/package/network/config/netifd/patches/003-ethtool.patch
@@ -0,0 +1,31 @@
+Index: netifd-2021-05-26-1eb0fafa/system-linux.c
+===================================================================
+--- netifd-2021-05-26-1eb0fafa.orig/system-linux.c
++++ netifd-2021-05-26-1eb0fafa/system-linux.c
+@@ -1578,6 +1578,8 @@ int system_vlandev_del(struct device *vl
+ 	return system_link_del(vlandev->ifname);
+ }
+ 
++#ifdef SPEED_400000
++// ugly hack to detect if we are running on an ancient kernel
+ static void
+ system_set_ethtool_settings(struct device *dev, struct device_settings *s)
+ {
+@@ -1628,6 +1630,7 @@ system_set_ethtool_settings(struct devic
+ 	ecmd.cmd = ETHTOOL_SSET;
+ 	ioctl(sock_ioctl, SIOCETHTOOL, &ifr);
+ }
++#endif
+ 
+ void
+ system_if_get_settings(struct device *dev, struct device_settings *s)
+@@ -1852,7 +1855,9 @@ system_if_apply_settings(struct device *
+ 		system_set_drop_unsolicited_na(dev, s->drop_unsolicited_na ? "1" : "0");
+ 	if (apply_mask & DEV_OPT_ARP_ACCEPT)
+ 		system_set_arp_accept(dev, s->arp_accept ? "1" : "0");
++#ifdef SPEED_400000
+ 	system_set_ethtool_settings(dev, s);
++#endif
+ }
+ 
+ int system_if_up(struct device *dev)
-- 
2.35.1

